### ISSUE TITLE
No way to pause auto updating?

### ISSUE THREAD

**Issue by keith009** [currency-cop/currency-cop#57]
I want to automatically update "holdings" every hour(or manually) but there isnt an option to do so anymore?

**Comment by leet0rz** [comment]
I miss this too, and all of sudden it doenst show any currency anymore, it goes back to zero. Whats up with this program right now? Its so buggy.

**Comment by nijikokun** [comment]
> I want to automatically update "holdings" every hour(or manually) but there isnt an option to do so anymore?

Yes, a pause feature is planned. The reason for the prices being updated constantly is to prevent what users ran into with the old version: rate limiting. Requests are fetched at an interval to prevent rate limiting. This not only affects your usage with the application but also elements on the site.
> I miss this too, and all of sudden it doenst show any currency anymore, it goes back to zero. Whats up with this program right now? Its so buggy.

Can you please let me know what you mean by "goes back to zero"? If you're talking about the "change" that's because it's the difference between this portfolio report build and the previous, check the drop down if you want historical statistics.

**Comment by leet0rz** [comment]
It goes back to not finding the currency at all, it shows my currency, then
the next update it shows nothing at all. It happens for days  at a time
even, its strange. thanks.

**Comment by leet0rz** [comment]
It seems that when u install this it doesnt get added in the standard
microsoft firewall, u have to manually go in and add this to exceptions.
Now it shows way more than before, however it does go to zero now and
again. I dont know if it loses connection to the api or whats going on.

**Comment by nijikokun** [comment]
@asdf337 depends on how many tabs you're tracking, please see this post for more information: https://www.pathofexile.com/forum/view-thread/1989935/page/60#p15643443

The latest release `beta.9` should reduce the `0` issue a lot, I have improved the matching mechanisms for portfolio tabs.

**Comment by leet0rz** [comment]
I am only tracking 1 tab as of now, my currency tab. Thanks for the info.
Thanks for continuing to update this amazing program, it has a lot of
potential.

**Comment by nijikokun** [comment]
> I am only tracking 1 tab as of now, my currency tab. Thanks for the info. Thanks for continuing to update this amazing program, it has a lot of potential.

I would like to see your logs if possible, they can be found under `%appdata%/currency-cop/`. If you decide to post them, use https://gist.github.com or pastebin.

**Comment by leet0rz** [comment]
first log:
https://pastebin.com/JL9yZmUR
second log:
https://pastebin.com/Rrkf0591

**Comment by leet0rz** [comment]
The currency cop tab just went back to zero now on the next update cycle it
did.

**Comment by leet0rz** [comment]
Here is a picture of the updates:
https://i.imgur.com/4BvqU4g.png

**Comment by nijikokun** [comment]
Are you sure that you're only fetching `1` tab?

Here it says you've been rate limited (fetching over 45 tabs per minute will cause this):

> [02:07:47] [WARN] [API] 429 status response for [https://www.pathofexile.com/character-window/get-stash-items]: {"error":{"code":3,"message":"Rate limit exceeded; You are requesting your character's items too frequently. Please try again later."}}

Also noticing a lot of issues where it cannot cache information:

> [21:40:52] [WARN] [Portfolio] File could not be loaded: ENOENT: no such file or directory, open 'C:\Users\Waffle\AppData\Roaming\currency-cop\Portfolios.db'

Are you running the application unpriviledged? This would make it unable to create the files, or have you been deleting those files?

**Comment by nijikokun** [comment]
Let's move our discussion over to #63

**Comment by leet0rz** [comment]
Maybe its because I have acquisition open too? It refreshes tabs all the
time, and a lot of them because of standard has endless tabs, might that be
it?

**Comment by nijikokun** [comment]
Yes @asdf337 that would certainly cause it, the latest version will better handle the rate-limiting

**Comment by leet0rz** [comment]
Thanks, something els I would love to see is a manually update button
instead of an automatic one. That would be awesome to see, for us
oldschoolers who have been using the older version for a long time.

**Comment by nijikokun** [comment]
> Thanks, something els I would love to see is a manually update button instead of an automatic one. That would be awesome to see, for us oldschoolers who have been using the older version for a long time.

It's not something that will happen. The pause functionality will be the new "refresh" in a sense. But the current architecture doesn't support the old way, and never will.
